### PR TITLE
Allow use of whois for offline users 

### LINKED ISSUE #1766: REQUEST: Offline support for /Whois and a question.
I have a request. 

Can offline support be added on /whois? So that we can look players up that are offline.

And also. 

What the permission to bypass 

    # Prevent the spawning of creatures.
    spawn:

If there is not, Can this be added or just something that can bypass it.

Note: I know. This is not an Issue. But i dont understand how "Pull request" works. Sorry

### PR DESCRIPTION
**Pull request by JRoy**
Fixes #1766

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/commands/Commandwhois.java [hunk 1]
```diff
@@ -30,38 +32,89 @@ public void run(final Server server, final CommandSource sender, final String co
         if (args.length < 1) {
             throw new NotEnoughArgumentsException();
         }
+        User player;
+        try {
+            final UUID uuid = UUID.fromString(args[0]);
+            player = ess.getUser(uuid);
+        }
+        catch (IllegalArgumentException ignored) {
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/commands/Commandwhois.java]: This code (and any that follow similarly) should be placed on the same line of code as the bracket to stay consistent with ess code style

**Review by pop4959 — CHANGES_REQUESTED** [review]
Looks pretty good overall to me. One thing I don't agree with though is adding an online/offline time to whois, because the seen command is already entirely designated to handling that specific task.

Also it wasn't explicitly clear what was wished for in the feature request, but this doesn't actually get all of the information that whois normally returns to the user. It shows what it can, which is still probably better than denying the command entirely for offline players.

Online

image

Offline

image

**Comment by Destitution-SC** [comment]
You could also add the mute/ban info to offline players for /whois

**Comment by JRoy** [comment]
@pop4959 Changes applied

**Review by darbyjack — APPROVED** [review]

**Review by darbyjack — DISMISSED** [review]
Just tested this locally and it looks to be working fine. This is a great feature to be implemented. Thanks for doing this!

**Review by pop4959 — CHANGES_REQUESTED** [review]
Just remove the online/offline text entirely -- it's not useful, as this is pretty obvious information, and it adds extra clutter.

Also, I'm not really sure if I agree with adding ban information to this command, as this duplicates even more behavior of `/seen`.

On the whole, the more I think about this issue the more I think this information should be put into `/seen` instead of `/whois`. It's basically the same command, except with the intent for being used to view information on offline players. Especially since half of the information in `/whois` doesn't even apply to offline users, this would make more sense and be less confusing in my opinion.

**Comment by JRoy** [comment]
@pop4959 As for ban information: the whois command already displays mute information so I felt that kind of fit. As for seen vs whois, you're right there is an information overlap. Let me look into that further

**Comment by pop4959** [comment]
> As for ban information: the whois command already displays mute information so I felt that kind of fit. As for seen vs whois, you're right there is an information overlap. Let me look into that further

That wasn't really my point there. If `/whois`' functionality is extended to support offline players as well, then of course it does make sense for it to show this sort of information. But remember that this command was, before this PR, not intended to be used by offline players.

The issue that is linked doesn't really elaborate on what they mean by offline whois, but if you want to be pedantic, it means that **all** functionality of whois should be available for offline players, which this PR **does not** do. While obviously having everything is likely not feasible to do currently due to limitations in the API, I don't think providing a half working implementation is the right way to go. It's just begging for further confusion, and further disappointment from people who want to see the information that is not shown.

If you read in between the lines, what probably makes more sense is to add additional information to `/seen` instead, which is the corresponding command for looking up offline players. This way, whois doesn't need to be muddied up with a half working offline implementation, but instead seen can show just the information that is relevant to offline players.

**Review by pop4959 — CHANGES_REQUESTED** [review]
I still think this should be better implemented by enhancing `/seen`. I might be willing to do this myself at some point.

**Comment by JRoy** [comment]
@pop4959 feel free, i'm not sure how i wanted to do this, you have push perms on my fork

**Comment by pop4959** [comment]
I'm not sure either, but it might be good to add with #3362 as well. Also worth asking on #1766 what they actually wanted out of this, so I've done that.

**Pull request closed without merging**